### PR TITLE
feat(newsletter-card): error state handling

### PR DESCRIPTION
**Pull request by DamianBrzezinskiHoA**
Appended:
-translations
-error card to handle with error state

### CODE REVIEW

File: libs/blog/newsletter/feature-newsletter/src/lib/newsletter-error/newsletter-error.component.html [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<div class="flex flex-col items-center justify-center text-center">
+  <div class="text-al-pink mb-2 mt-5 aspect-square h-20">
```
Comment (DamianBrzezinskiHoA) [inline libs/blog/newsletter/feature-newsletter/src/lib/newsletter-error/newsletter-error.component.html:2]: FYI: I decided NOT TO use our `al-icon` component because in this case we have to get this icon on init. Al icon gets svgs dynamically so when connection is broken, the icon will not be gotten (what is obvious). What is more, I also decided not to create another component for this single icon to keep to the `KISS` rule.

File: libs/blog/newsletter/feature-newsletter/src/lib/newsletter-error/newsletter-error.component.html [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        height="100%"
+        viewBox="0 0 12 12"
+        width="100%"
+        xmlns="http://www.w3.org/2000/svg"
+      >
+        <path
+          d="M10.2427 0.814333L6 5.05733L1.75734 0.814333C1.627 0.684 1.416 0.684 1.286 0.814333L0.814669 1.28567C0.684336 1.416 0.684336 1.627 0.814669 1.757L5.05734 6L0.814669 10.243C0.684669 10.3733 0.684669 10.5843 0.814669 10.7143L1.286 11.1857C1.41634 11.316 1.62734 11.316 1.75734 11.1857L6 6.94267L10.2427 11.1857C10.373 11.316 10.584 11.316 10.714 11.1857L11.1853 10.7143C11.3157 10.584 11.3157 10.373 11.1853 10.243L6.94267 6L11.1853 1.757C11.3153 1.62667 11.3153 1.41567 11.1853 1.28567L10.714 0.814333C10.5837 0.684 10.3727 0.684 10.2427 0.814333Z"
+          fill="currentColor"
+        />
+      </svg>
```
Comment (mateuszbasinski) [inline libs/blog/newsletter/feature-newsletter/src/lib/newsletter-error/newsletter-error.component.html:15]: I don't think it is a good idea to put here svg tag

File: libs/blog/newsletter/feature-newsletter/src/lib/newsletter-error/newsletter-error.component.html [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    </div>
+    <h2 class="my-2 text-2xl font-bold">
+      {{ t('title') }}
+    </h2>
+    <p class="my-5">
+      {{ t('description') }}
+    </p>
+    <p class="text-al-blue my-5 cursor-pointer" (click)="onReturnClick()">
```
Comment (mateuszbasinski) [inline libs/blog/newsletter/feature-newsletter/src/lib/newsletter-error/newsletter-error.component.html:23]:
```suggestion
    <p class="text-al-blue my-5 cursor-pointer" (click)="returnClicked.emit()">
```

File: libs/blog/newsletter/feature-newsletter/src/lib/newsletter-error/newsletter-error.component.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  imports: [IconComponent, TranslocoDirective],
+  templateUrl: 'newsletter-error.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class NewsletterErrorComponent {
+  returnClicked = output<void>();
+
+  onReturnClick(): void {
+    this.returnClicked.emit();
+  }
```
Comment (mateuszbasinski) [inline libs/blog/newsletter/feature-newsletter/src/lib/newsletter-error/newsletter-error.component.ts:18]: remove

File: libs/blog/newsletter/feature-newsletter/src/lib/newsletter-error/newsletter-error.component.spec.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { NewsletterErrorComponent } from './newsletter-error.component';
+
+describe('NewsletterErrorComponent', () => {
```
Comment (mateuszbasinski) [inline libs/blog/newsletter/feature-newsletter/src/lib/newsletter-error/newsletter-error.component.spec.ts:5]: Consider to write some tests

File: libs/blog/newsletter/feature-newsletter/src/lib/newsletter-error/newsletter-error.component.html (hunk 2 above)
Comment (DamianBrzezinskiHoA) [reply libs/blog/newsletter/feature-newsletter/src/lib/newsletter-error/newsletter-error.component.html:15]: We can use `al-icon` but as I mentioned, SVG will not be gotten. Corrected

File: libs/blog/newsletter/feature-newsletter/src/lib/newsletter-error/newsletter-error.component.spec.ts (hunk 5 above)
Comment (DamianBrzezinskiHoA) [reply libs/blog/newsletter/feature-newsletter/src/lib/newsletter-error/newsletter-error.component.spec.ts:5]: AFAIK we are not going to write unit tests for simple dumb components.

File: libs/blog/newsletter/feature-newsletter/src/lib/feature-newsletter/newsletter.component.ts [hunk 6]
```diff
@@ -49,13 +47,14 @@ export class NewsletterComponent {
...
-    if (this._newsletterStore.loading() === 'success') {
+  private readonly _onResult = effect(() => {
+    const loadingState = this._newsletterStore.loading();
+    if (loadingState === 'success' || loadingState === 'error') {
       this._router.navigate(
         this._localizeRouter.translateRoute(['/newsletter']) as string[],
         {
           queryParams: {
-            nm: 'confirmed',
+            nm: loadingState === 'success' ? 'confirmed' : 'not-confirmed',
```
Comment (DDonochVA) [inline libs/blog/newsletter/feature-newsletter/src/lib/feature-newsletter/newsletter.component.ts:57]: we navigate only for tracking purposes and don't wanna navigate user to that page on error. 
Error should be displayed down below the form.

File: libs/blog/newsletter/feature-newsletter/src/lib/feature-newsletter/newsletter.component.html [hunk 7]
```diff
@@ -38,16 +38,20 @@ <h2 class="text-4xl font-bold">{{ t('title') }}</h2>
...
+                  >
+                    <div
+                      class="bg-al-pink hidden h-[6px] w-[6px] rounded-sm"
+                    ></div>
+                  </div>
+                  <span class="before:text-al-pink before:content-['*']">
+                    {{ t('consent') }}
+                  </span>
+                } @else {
+                  <p class="font-bold">{{ t('error') }}</p>
```
Comment (DamianBrzezinskiHoA) [inline libs/blog/newsletter/feature-newsletter/src/lib/feature-newsletter/newsletter.component.html:53]: I decided to use `font-bold` instead `text-al-primary` class because `text-al-primary ` does not provide good contrast

File: libs/blog/newsletter/feature-newsletter/src/lib/feature-newsletter/newsletter.component.ts (hunk 6 above)
Comment (DamianBrzezinskiHoA) [reply libs/blog/newsletter/feature-newsletter/src/lib/feature-newsletter/newsletter.component.ts]: Corrected

**Comment by DDonochVA** [comment]
image
checkbox should still be visible on the error state

File: libs/blog/newsletter/data-access/src/lib/state/newsletter.store.ts [hunk 8]
```diff
@@ -15,6 +15,7 @@ const initialState: NewsletterState = {
 };
 
 export const NewsletterStore = signalStore(
+  { providedIn: 'root' },
```
Comment (DDonochVA) [inline libs/blog/newsletter/data-access/src/lib/state/newsletter.store.ts:18]: remove

**Comment by DamianBrzezinskiHoA** [comment]
> image checkbox should still be visible on the error state


I was wondering about it. In this case, the user gave his consent earlier, so I think we shouldn't ask him for this once more time. To my mind, asking twice for somebody's consetnt doen not make sense.

**Comment by DDonochVA** [comment]
> I was wondering about it. In this case, the user gave his consent earlier, so I think we shouldn't ask him for this once more time. To my mind, asking twice for somebody's consetnt doen not make sense.

I've never had this kind of experience on the web.  Imo form should be persistent

File: libs/blog/newsletter/feature-newsletter/src/lib/feature-newsletter/newsletter.component.ts [hunk 9]
```diff
@@ -32,9 +32,7 @@ import { NewsletterStore } from '@angular-love/data-access';
     TranslocoDirective,
   ],
   templateUrl: './newsletter.component.html',
-  styleUrl: './newsletter.component.scss',
   changeDetection: ChangeDetectionStrategy.OnPush,
-  providers: [NewsletterStore],
```
Comment (DDonochVA) [inline libs/blog/newsletter/feature-newsletter/src/lib/feature-newsletter/newsletter.component.ts:37]: why did you remove this? restore

**Review by DDonochVA — APPROVED** [review]
